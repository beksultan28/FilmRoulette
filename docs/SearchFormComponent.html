<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: SearchFormComponent</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: SearchFormComponent</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>SearchFormComponent<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="SearchFormComponent"><span class="type-signature"></span>new SearchFormComponent<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Search form componentPerforms search on search form submit,state change and first load
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_search-form_index.jsx.html">components/search-form/index.jsx</a>, <a href="components_search-form_index.jsx.html#line9">line 9</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    
        <h3 class="subsection-title">Extends</h3>

        


    <ul>
        <li>Component</li>
    </ul>


    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id=".handleSearch"><span class="type-signature">(static) </span>handleSearch<span class="signature">(inputs)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Handles search form submitPerforms search with form dataand changes location to the search page
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>inputs</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_search-form_index.jsx.html">components/search-form/index.jsx</a>, <a href="components_search-form_index.jsx.html#line79">line 79</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id=".handleSearchBy"><span class="type-signature">(static) </span>handleSearchBy<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    searchBy input change event handlerupdates searchBy state with input value
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_search-form_index.jsx.html">components/search-form/index.jsx</a>, <a href="components_search-form_index.jsx.html#line96">line 96</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id=".updateRequest"><span class="type-signature">(static) </span>updateRequest<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Parses url and updated statesthen performs search for films with new states
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_search-form_index.jsx.html">components/search-form/index.jsx</a>, <a href="components_search-form_index.jsx.html#line59">line 59</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="DetailComponent.html">DetailComponent</a></li><li><a href="ResultsListComponent.html">ResultsListComponent</a></li><li><a href="ResultsSortComponent.html">ResultsSortComponent</a></li><li><a href="SearchFormComponent.html">SearchFormComponent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#apiKey">apiKey</a></li><li><a href="global.html#detailItem">detailItem</a></li><li><a href="global.html#expectedResult">expectedResult</a></li><li><a href="global.html#expectedResults">expectedResults</a></li><li><a href="global.html#FooterComponent">FooterComponent</a></li><li><a href="global.html#getFilmApi">getFilmApi</a></li><li><a href="global.html#getFilmById">getFilmById</a></li><li><a href="global.html#getRecommendations">getRecommendations</a></li><li><a href="global.html#getRecommendationsApi">getRecommendationsApi</a></li><li><a href="global.html#recieveFilmById">recieveFilmById</a></li><li><a href="global.html#recieveFilms">recieveFilms</a></li><li><a href="global.html#ResultsItemComponent">ResultsItemComponent</a></li><li><a href="global.html#searchApi">searchApi</a></li><li><a href="global.html#searchBy">searchBy</a></li><li><a href="global.html#searchFilms">searchFilms</a></li><li><a href="global.html#searchResults">searchResults</a></li><li><a href="global.html#sortBy">sortBy</a></li><li><a href="global.html#sortFilms">sortFilms</a></li><li><a href="global.html#updateQuery">updateQuery</a></li><li><a href="global.html#updateSearchBy">updateSearchBy</a></li><li><a href="global.html#updateSortBy">updateSortBy</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sat Oct 28 2017 22:06:51 GMT+0600 (Central Asia Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>