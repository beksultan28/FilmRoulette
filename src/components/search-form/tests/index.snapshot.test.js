import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import SearchFormComponent from '../index.jsx';

const searchFilms = jest.fn();
const updateQuery = jest.fn();
const updateSearchBy = jest.fn();
const handleSubmit = jest.fn();
const props = {
  searchQuery: 'Test',
  searchBy: 'movie',
  sortBy: 'release_date',
  match: {
    params: {
      query: 'Test',
    },
  },
  history: [],
  searchFilms,
  updateQuery,
  updateSearchBy,
  handleSubmit,
};

const renderer = new ShallowRenderer();
renderer.render(
  <SearchFormComponent
    {...props}
  />,
);
const searchForm = renderer.getRenderOutput();

test('SearchFormComponent renders correctly', () => {
  expect(searchForm).toMatchSnapshot();
});
