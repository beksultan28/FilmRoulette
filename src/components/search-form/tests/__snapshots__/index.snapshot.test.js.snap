// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFormComponent renders correctly 1`] = `
<form
  className="search-form"
  onSubmit={undefined}
>
  <h2
    className="search-title"
  >
    Find your movie
  </h2>
  <Field
    className="search-query"
    component="input"
    name="searchQuery"
    placeholder="Type there movie title or director"
    type="text"
  />
  <div
    className="clearfix"
  >
    <div
      className="search-fields"
    >
      <span>
        Search 
      </span>
      <span>
        <input
          checked={true}
          id="ch-movie"
          name="searchBy"
          onChange={[Function]}
          type="radio"
          value="movie"
        />
        <label
          htmlFor="ch-movie"
        >
           movie 
        </label>
      </span>
      <span>
        <input
          checked={false}
          id="ch-tv"
          name="searchBy"
          onChange={[Function]}
          type="radio"
          value="tv"
        />
        <label
          htmlFor="ch-tv"
        >
           tv 
        </label>
      </span>
    </div>
    <button
      className="btn btn-red search-button"
      type="submit"
    >
      search
    </button>
  </div>
</form>
`;
